src/04-spin-boson-model.md

```diff
@@ -0,0 +1,800 @@
+    
+    # map from QuTiP number states to |n,±> states
+    possible_ns = range(0, max_bosons+1)
+    possible_ms = ["+","-"]
+    nm_list = [(n,m) for (n,m) in product(possible_ns, possible_ms)]
+    
+    # only do parity extraction if a valid parity is being used
+    if (parity==1) | (parity==-1):
+        p           = np.where(P.diag()==parity)[0]
+        
+        two_state   = two_state.extract_states(p)
+        bosons      = bosons.extract_states(p)
+        number      = number.extract_states(p)
+        interaction = interaction.extract_states(p)
+        nm_list     = [nm_list[i] for i in p]
+    
+    
+    return two_state, bosons, interaction, number, nm_list
+```
+
+We are now in a position to compare the energy level plots for the different parity universes side by side.
+
+```python
+# ODD PARITY
+
+two_state, bosons, interaction, number, nm_list = make_operators(max_bosons=4, parity=-1)
+
+df_odd = make_df_for_energy_scan("$\Delta E$", -4, 4, 201, two_state.shape[0])
+
+for i, row in df_odd.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df_odd.iloc[i,1:] = evals 
+```
+
+```python
+# EVEN PARITY
+
+two_state, bosons, interaction, number, nm_list = make_operators(max_bosons=4, parity=1)
+
+df_even = make_df_for_energy_scan("$\Delta E$", -4, 4, 201, two_state.shape[0])
+
+for i, row in df_even.iterrows():
+    H =  row["$\Delta E$"]*two_state + 1*bosons + 0.2*interaction
+    evals, ekets = H.eigenstates()
+    df_even.iloc[i,1:] = evals 
+```
+
+```python
+fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(15,6), sharey=True)
+
+fig.suptitle(f"Stationary states for {H_latex}    ($\omega=1$, $U=0.2$)")
+
+df_odd.plot(x="$\Delta E$",ylim=[-0.5,5.5],legend=False, 
+        title="Odd parity     (Fig 9)",  ax=axes[0]);
+
+df_even.plot(x="$\Delta E$",ylim=[-0.5,5.5],legend=False, 
+        title="Even parity      (Fig 10)",  ax=axes[1]);
+
+axes[0].set_ylabel("Energy");
+
+```
+
+Figs 9 and 10 show us the energy of the stationary states for the odd and even parity states respectively. As we expected, all levels now anti-cross when they approach each other because in each universe there are no forbidden interactions.
+
+We can now see that the appearance of resonances in Fig 1 at $\Delta E = 2\omega, 4\omega$ etc was in a sense an illusion and that only the odd resonances $\Delta E = \omega, 3\omega, 5\omega$ are physical.
+
+Good news - with separate parity universes, we can run all simulations much faster because we can remove half the states from consideration 🙌. This includes, of course, simulating the anti-crossing that inspired the down conversion ideas in Fig 3. Let's get to it!
+
+
+## 4.5 - Down conversion
+
+
+We've seen several signs that when $\Delta E \approx 3\omega$ (anti-crossing in Fig 10 and also Fig 3) we can expect down conversation, i.e. $|0,+> \rightarrow |3,->$. Let's simulate and see if we are correct.
```

Comment:
Do you mean to say "down conversation" here?

Reply:
This point is generally clear to me. I suppose we have deltaE = 3 bosons -- right?